fw/vhost.c

```diff
@@ -2572,6 +2615,188 @@ tfw_vhost_cfgclean(void)
 	tfw_global.hdr_via = s_hdr_via_dflt;
 }
 
+#if defined(PRINT_CONFIG)
+void
+printFrang(const char *tab, const FrangVhostCfg *frang)
+{
+	int i;
+
+	if (frang) {
+		T_LOG_NL("%sfrang_limits {\n", tab);
+		T_LOG_NL("%s   http_methods_mask=%lu\n",
+			 tab, frang->http_methods_mask);
+		T_LOG_NL("%s   http_body_len=%lu\n",
+			 tab, frang->http_body_len);
+		T_LOG_NL("%s   http_uri_len=%i\n", tab,
+			 frang->http_uri_len);
+		if (!frang->http_ct_vals) {
+			T_LOG_NL("%s   http_ct_vals=NULL\n", tab);
+		}
+		else {
+			T_LOG_NL("%s   http_ct_vals=%s\n", tab,
+				 frang->http_ct_vals->data);
+		}
+
+		if (!frang->http_resp_code_block) {
+			T_LOG_NL("%s   FrangHttpRespCodeBlock: Empty\n", tab);
+		}
+		else {
+			T_LOG_NL("%s   FrangHttpRespCodeBlock:\n", tab);
+			for (i = 0; i < 512; ++i) {
+				if (test_bit(HTTP_CODE_BIT_NUM(i),
+					     frang->http_resp_code_block->codes))
+					T_LOG_NL("%s   %i\n", tab, i);
+			}
+		}
+
+		T_LOG_NL("%s   http_ct_required=%s\n", tab,
+			 frang->http_ct_required ? "true" : "false");
+		T_LOG_NL("%s   http_strict_host_checking=%s\n", tab,
+			 frang->http_strict_host_checking ? "true" : "false");
+		T_LOG_NL("%s   http_trailer_split=%s\n", tab,
+			 frang->http_trailer_split ? "true" : "false");
+		T_LOG_NL("%s   http_method_override=%s\n", tab,
+			 frang->http_method_override ? "true" : "false");
+
+		T_LOG_NL("%s}\n", tab);
+	}
+}
+
+
+void printBasicStr(BasicStr *str)
+{
+	char new_str[128];
+	int len;
+
+	memset(new_str, 0, sizeof(new_str));
+	len = str->len < 128 ? str->len : 127;
+	memcpy(new_str, str->data, len);
+	T_LOG_NL("%s\n", new_str);
+}
+
+
+static void
+tfw_cfgop_location_print(TfwLocation *loc)
+{
+        int i;
+         TfwHdrMods *h_mods = &loc->mod_hdrs[TFW_VHOST_HDRMOD_RESP];
+
+        if (loc->arg) {
```

Comment:
```suggestion
        if (!loc->arg)
                return;
```
Please do your best to reduce nesting in the code